debezium-core/src/main/java/io/debezium/transforms/UnwrapFromEnvelope.java

```diff
@@ -164,13 +180,31 @@ public void configure(final Map<String, ?> configs) {
 
     @Override
     public R apply(final R record) {
+        Envelope.Operation operation;
         if (record.value() == null) {
             if (dropTombstones) {
                 logger.trace("Tombstone {} arrived and requested to be dropped", record.key());
                 return null;
             }
+            operation = Envelope.Operation.DELETE;
+            if (addDebeziumOperationHeader) {
+                record.headers().addString(DEBEZIUM_OPERATION_HEADER_KEY, operation.toString());
+            }
             return record;
         }
+
+        if (addDebeziumOperationHeader) {
+            try {
+                operation = Envelope.Operation.forCode(
+                        ((Struct) record.value()).get("op").toString()
+                );
+                record.headers().addString(DEBEZIUM_OPERATION_HEADER_KEY, operation.code());
+            }
+            catch (NullPointerException exception) {
```

Comment:
Catching the NPE seems a bit fishy. I'd rather suggest to check for presence of the "op" field upfront.